src/common/tls.c

```diff
@@ -0,0 +1,191 @@
+/**
+ * (C) Copyright 2016-2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+/**
+ * It implements thread-local storage (TLS) for DAOS.
+ */
+#include <pthread.h>
+#include <daos/tls.h>
+
+/* The array remember all of registered module keys on one node. */
+struct daos_module_key *daos_module_keys[DAOS_MODULE_KEYS_NR] = { NULL };
```

Comment:
Should be static again and you don't need to initliase it.
```suggestion
static struct daos_module_key *daos_module_keys[DAOS_MODULE_KEYS_NR];
```

Reply:
sure